shell/platform/darwin/macos/README.md

```diff
@@ -0,0 +1,50 @@
+# Flutter macOS Embedder
+
+This directory contains files specific to the Flutter macOS embedder. The
+embedder is built as a macOS framework that is linked into the target
+application. The framework is composed both of macOS-specific code and code
+shared with the iOS embedder. These can be found in:
+```
+flutter/shell/platform/darwin/common/framework
+flutter/shell/platform/darwin/macos/framework
+```
+
+Additionally, the framework relies on utility code shared across all embedders,
+which is found in:
+```
+flutter/shell/platform/common
+```
+
+To learn more, see the [Engine architecture wiki](https://github.com/flutter/flutter/wiki/The-Engine-architecture).
+
+## Building
+
+Building all artifacts required for macOS occurs in two steps:
+* Host binary build: Builds tooling used by the Flutter tool to build Flutter
+  applications targeting macOS.
+* macOS Target build: Builds the framework that implements the macOS Flutter
+  embedder and exposes public API such as `FlutterViewController` used by
+  Flutter applications.
+
+To build the macOS embedder from the `src/flutter` directory, run the following
+commands:
+```sh
+# Perform the host build.
+./tools/gn --unopt --no-goma
+autoninja -C ../out/host_debug_unopt      
+
+# Perform the macOS target build.
+./tools/gn --unopt --mac --no-goma
+autoninja -C ../out/mac_debug_unopt
+```
+Builds are architecture-specific, and can be controlled by specifying
+`--mac-cpu=arm64` or `--mac-cpu=x64` (default) when invoking `gn`.
+
+Googlers can remove `--no-goma` to make use of the Goma distributed compile
+service.
+
+## Testing
+
+The macOS-specific host tests are built into `flutter_desktop_darwin_unittests`.
```

Comment:
Should this provide an example of the `run_tests.py` command? Consider adding something like:

````suggestion
The macOS-specific host tests are built into `flutter_desktop_darwin_unittests`:

```sh
# Run macOS embedder tests.
testing/run_tests.py --type engine --engine-filter flutter_desktop_darwin_unittests
```
````

Reply:
I'll add this as a separate section and point to the other readme. I suspect most people debugging will want to run the binary, either under a debugger, or directly with specific flags. I don't think I've used run_tests.py in years, but it's definitely useful for running broader swaths of tests as needed.

Reply:
Added a link to the engine testing wiki.